modules/inflowwind/src/InflowWind_Subs.f90

```diff
@@ -1681,63 +1666,50 @@ SUBROUTINE InflowWind_GetSpatialAverage( Time, InputData, p, x, xd, z, OtherStat
-   REAL(ReKi),                               INTENT(  OUT)  :: MeanVelocity(3)   !< at InputPosition
-      
+   REAL(ReKi),                               INTENT(  OUT)  :: MeanVelocity(3)   !< at InputPosition      
    INTEGER(IntKi),                           INTENT(  OUT)  :: ErrStat           !< Error status of the operation
    CHARACTER(*),                             INTENT(  OUT)  :: ErrMsg            !< Error message if ErrStat /= ErrID_None
+   
+   CHARACTER(*),PARAMETER     :: RoutineName = "InflowWind_GetSpatialAverage"
+   INTEGER(IntKi)             :: ErrStat2
+   CHARACTER(ErrMsgLen)       :: ErrMsg2   
+   INTEGER(IntKi)             :: I
+   REAL(ReKi), ALLOCATABLE    :: acc(:,:)  ! acceleration, unallocated so it won't be calculated
 
-
-      ! Local variables
-   INTEGER(IntKi)                                           :: I                   !< Generic counters
-
-
-      ! Temporary variables for error handling
-   INTEGER(IntKi)                                           :: ErrStat2
-   CHARACTER(ErrMsgLen)                                     :: ErrMsg2            ! temporary error message
-   LOGICAL,                PARAMETER                        :: FillWrOut = .false.
-
-
-
-      ! Initialize ErrStat
    ErrStat  = ErrID_None
    ErrMsg   = ""
    MeanVelocity = 0.0_ReKi
    
-   !-----------------------------------------------------------------------
-   !  Points coordinate transforms from to global to aligned with hub orientation
-   !-----------------------------------------------------------------------
+   !----------------------------------------------------------------------------
+   ! Points coordinate transforms from to global to aligned with hub orientation
+   !----------------------------------------------------------------------------
+
    m%u_Avg%HubPosition    = InputData%HubPosition
    m%u_Avg%HubOrientation = InputData%HubOrientation
    
-   do i=1,IfW_NumPtsAvg
-      m%u_Avg%PositionXYZ(:,i) = matmul(InputData%HubOrientation,p%PositionAvg(:,i)) + InputData%HubPosition
+   do i = 1, IfW_NumPtsAvg
+      m%u_Avg%PositionXYZ(:,i) = InputData%HubPosition + &
+                                 matmul(InputData%HubOrientation,p%PositionAvg(:,i))
    end do
 
-   CALL CalculateOutput( Time, m%u_Avg, p, x, xd, z, OtherStates, m%y_Avg, m, FillWrOut, ErrStat2, ErrMsg2 )
-      CALL SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg, RoutineName )
+   !----------------------------------------------------------------------------
+   ! Calculate wind velocities at average positions
+   !----------------------------------------------------------------------------
 
-   do i=1,IfW_NumPtsAvg
-      MeanVelocity = MeanVelocity + m%y_Avg%VelocityUVW(:,i)
-   end do
-   MeanVelocity = MeanVelocity / REAL(IfW_NumPtsAvg,ReKi)
-        
-   
-   RETURN
+   CALL IfW_FlowField_GetVelAcc(p%FlowField, 0, Time, m%u_Avg%PositionXYZ, &
```

Comment:
This is a very important function, but somehow I didn't see during the review, I probably missed it. Where is it defined?

Reply:
It's defined at the top of `IfW_FlowField.f90` which was added when InflowWind was restructured (#1516). The FlowFieldType is initialized in InflowWind and contains data for all of the wind types. The `GetVelAcc` function can then be used to query the velocity/acceleration at any point regardless of which wind type is being used.